pkg/checkoutinfo/checkoutinfo.go

```diff
@@ -0,0 +1,229 @@
+	Dir() string
+	URL() string
+}
+
+type CheckoutInfo struct {
+	auth    *authentication.Auth
+	cfg     configurer
+	project projecter
+}
+
+var ErrBuildscriptNotExist = errors.New("Build script does not exist")
+
+type ErrInvalidCommitID struct {
+	CommitID string
+}
+
+func (e ErrInvalidCommitID) Error() string {
+	return "invalid commit ID"
+}
+
+func New(auth *authentication.Auth, cfg configurer, project projecter) *CheckoutInfo {
+	return &CheckoutInfo{auth, cfg, project}
+}
+
+func (c *CheckoutInfo) Owner() string {
+	return c.project.Owner()
+}
+
+func (c *CheckoutInfo) Name() string {
+	return c.project.Name()
+}
+
+func (c *CheckoutInfo) Branch() string {
+	return c.project.BranchName()
+}
+
+func (c *CheckoutInfo) CommitID() (strfmt.UUID, error) {
+	if c.cfg.GetBool(constants.OptinBuildscriptsConfig) {
+		if script, err := c.BuildScript(); err == nil {
+			commitID, err2 := script.CommitID()
+			if err2 != nil {
+				return "", errs.Wrap(err, "Could not get commit ID from build script")
+			}
+			return commitID, nil
+		} else if !errors.Is(err, ErrBuildscriptNotExist) {
+			return "", errs.Wrap(err, "Could not get build script")
+		}
+		// Fall back on activestate.yaml.
+	}
+
+	// Read from activestate.yaml.
+	commitID := c.project.LegacyCommitID()
+	if !strfmt.IsUUID(commitID) {
+		return "", &ErrInvalidCommitID{commitID}
+	}
+
+	return strfmt.UUID(commitID), nil
+}
+
+func (c *CheckoutInfo) BuildScript() (*buildscript.BuildScript, error) {
+	if !c.cfg.GetBool(constants.OptinBuildscriptsConfig) {
+		bp := buildplanner.NewBuildPlannerModel(c.auth)
+		script, err := bp.GetBuildScript(c.Owner(), c.Name(), c.Branch(), c.project.LegacyCommitID())
+		if err != nil {
+			return nil, errs.Wrap(err, "Could not get remote build script")
+		}
+		return script, nil
+	}
+
+	path := filepath.Join(c.project.Dir(), constants.BuildScriptFileName)
+
+	data, err := fileutils.ReadFile(path)
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			return nil, errs.Pack(err, ErrBuildscriptNotExist)
+		}
+		return nil, errs.Wrap(err, "Could not read build script from file")
+	}
+
+	script, err := buildscript.Unmarshal(data)
+	if err != nil {
+		return nil, errs.Wrap(err, "Could not unmarshal build script")
+	}
+
+	return script, nil
+}
+
+func (c *CheckoutInfo) SetCommitID(commitID strfmt.UUID) error {
+	// Update commitID in activestate.yaml.
+	logging.Debug("Updating commitID in activestate.yaml")
+	if err := c.project.SetLegacyCommit(commitID.String()); err != nil {
+		return errs.Wrap(err, "Could not set commit ID")
+	}
+
+	if !c.cfg.GetBool(constants.OptinBuildscriptsConfig) {
+		return nil // buildscripts are not enabled, so nothing more to do
+	}
+
+	// Update commitID in Project field of build script.
+	logging.Debug("Updating commitID in buildscript")
+	buildscriptPath := filepath.Join(c.project.Dir(), constants.BuildScriptFileName)
+
+	if !fileutils.FileExists(buildscriptPath) {
+		return c.InitializeBuildScript(commitID)
+	}
```

Comment:
Please error out with the "run state reset LOCAL" error. This file should exist if opted in, it not existing is an error condition. Working around this magically like this is what we did originally and it caused a lot of issues.